hammr/commands/image/image.py

```diff
@@ -271,117 +270,142 @@ def help_deploy(self):
-                                    "Do you really want to cancel image with id " + str(cancelImage.dbId)):
-                        self.api.Users(self.login).Appliances(
-                            generics_utils.extract_id(cancelImage.applianceUri)).Images(
-                            cancelImage.dbId).Status.Cancel()
+            printer.out("Searching image with id [" + do_args.id + "] ...")
+            images = self.get_all_images()
+            if images is None :
+                raise ValueError("No image found")
+
+            table = Texttable(800)
+            table.set_cols_dtype(["t", "t", "t", "t", "t", "t", "t", "t", "t"])
+            table.header(["Id", "Name", "Version", "Rev.", "Format", "Created", "Size", "Compressed", "Status"])
+            cancel_image = None
+            for image in images:
+                if str(image.dbId) == str(do_args.id):
+                    img_status = self.get_image_status(image.status)
+                    table.add_row([image.dbId, image.name, image.version, image.revision, image.targetFormat.name,
+                                   image.created.strftime("%Y-%m-%d %H:%M:%S"), size(image.size),
+                                   "X" if image.compress else "", img_status])
+                    print table.draw() + "\n"
+                    cancel_image = image
+            if cancel_image is None or cancel_image.status.complete or cancel_image.status.cancelled:
+                raise ValueError("Image not being generated, impossible to canceled")
+
+            if cancel_image is not None:
+                if generics_utils.query_yes_no(
+                                "Do you really want to cancel image with id " + str(cancel_image.dbId)):
+
+                    if is_uri_based_on_appliance(cancel_image.uri):
+                        appliance_id = extract_appliance_id(cancel_image.uri)
+                        self.api.Users(self.login).Appliances(appliance_id).Images(cancel_image.dbId).Status.Cancel()
                         printer.out("Image Canceled", printer.OK)
-                else:
-                    printer.out("Image not found", printer.ERROR)
 
+                    elif is_uri_based_on_scan(cancel_image.uri):
+                        scanned_instance_id = extract_scannedinstance_id(cancel_image.uri)
+                        scan_id = extract_scan_id(cancel_image.uri)
+                        self.api.Users(self.login).Scannedinstances(scanned_instance_id).Scans(scan_id).\
+                            Images(None, cancel_image.dbId).Status.Cancel()
+                        printer.out("Image Canceled", printer.OK)
+
+            else:
+                printer.out("Image not found", printer.ERROR)
 
         except ArgumentParserError as e:
             printer.out("ERROR: In Arguments: " + str(e), printer.ERROR)
             self.help_delete()
+            return 2
+        except ValueError as e:
+            printer.out(str(e), printer.ERROR)
```

Comment:
See my comment on get_all_images. I think here the user will see two messages "No images available" and "No image found". Maybe you should raise the exception inside "get_all_images" and avoid raising another one here.
And maybe we can use a specific exception instead of the "ValueError" one.

Reply:
After a discussion offline, we could not find something better than ValueError